### DISCUSSION TITLE
Very slow nested queries

### DISCUSSION THREAD

**Discussion by PhilBookst** [prisma/prisma#9445] [Q&A]
Hey, i'm having trouble using nested queries - my db and server are hosted in the same city so it's not because of latency

My schema looks like this (not the full schema if it matters):

```
model Profile {
  id              String?     @unique
  username        String      @id
  is_private      Boolean?
  is_verified     Boolean?
  full_name       String?
  biography       String?
  profile_pic_url String?
  post_amount     Int?
  follower        Int?
  following       Int?
  imageTags       ImageTags[]
  nodes           Image[]

  SearchProfiles SearchProfiles[]
  @@map("profile")
}

model Image {
  id              String           @id
  thumbnail       String
  shortcode       String           @unique
  liked           Int
  comment_count   Int
  engagement_rate Float?
  day             Int?
  hour            Int?
  date            String
  caption         String?

  profile         Profile?         @relation(fields: [profileUsername], references: [username])
  profileUsername String?

  hashtag         Hashtag?         @relation(fields: [hashtagName], references: [name])
  hashtagName     String?
  location        Location?        @relation(fields: [locationId], references: [id])
  locationId      String?
  imageHashtags   ImageHashtags[]
  imageTags       ImageTags[]
  imageLocations  ImageLocations[]

  @@map("image")
}
```

if i run this query, it takes about 4-5 seconds to give me the results

```
const user = await prisma.profile.findUnique({
    where: {
      username,
    },
    include: {
      nodes: true,
    },
  });
```

if i run it without the include it takes around 200ms

```
const user = await prisma.profile.findUnique({
     where: {
      username,
    },
  });
```

the profile table has around 370k rows and and image table around 620k

**Comment by ryands17** [comment]
@PhilBookst 👋 

It would be great if you could create an issue [here](https://github.com/prisma/prisma/issues/new?assignees=&labels=kind%2Fbug&template=bug_report.yml) with all the logs and reproduction steps so that we can look into this 🙂

**Comment by ludralph** [comment]
Hi there,

To keep our discussions organized and focused on the most relevant topics, we’re reviewing and tidying up our backlog. As part of this process, we’re closing discussions that haven’t had any recent activity and appear to be outdated.

If this discussion is still important to you or unresolved, we’d love to hear from you! Feel free to reopen it or start a new one with updated details.

For more details about our priorities and vision for the future of Prisma ORM, check out our latest blog post: https://www.prisma.io/blog/prisma-orm-manifesto.

Thank you for your understanding and being part of the community!
